src/circuit/gadget/ecc/chip/mul_fixed/short.rs

```diff
@@ -0,0 +1,317 @@
+                    Some(scalar_fixed),
+                )?;
+                value_commit_v.mul(layouter.namespace(|| "mul by zero"), &scalar_fixed)?
+            };
+            constrain_equal(
+                chip.clone(),
+                layouter.namespace(|| "mul by zero"),
+                base_val,
+                scalar_fixed,
+                result,
+            )?;
+        }
+
+        // Random [a]B
+        {
+            let scalar_fixed_short = pallas::Scalar::from_u64(rand::random::<u64>());
+            let mut sign = pallas::Scalar::one();
+            if rand::random::<bool>() {
+                sign = -sign;
+            }
+            let scalar_fixed_short = sign * &scalar_fixed_short;
+
+            let result = {
+                let scalar_fixed_short = ScalarFixedShort::new(
+                    chip.clone(),
+                    layouter.namespace(|| "random short scalar"),
+                    Some(scalar_fixed_short),
+                )?;
+
+                value_commit_v.mul(layouter.namespace(|| "random [a]B"), &scalar_fixed_short)?
+            };
+            constrain_equal(
+                chip.clone(),
+                layouter.namespace(|| "random [a]B"),
+                base_val,
+                scalar_fixed_short,
+                result,
+            )?;
+        }
+
+        // [2^64 - 1]B
+        {
+            let scalar_fixed_short = pallas::Scalar::from_u64(0xFFFF_FFFF_FFFF_FFFFu64);
+
+            let result = {
+                let scalar_fixed_short = ScalarFixedShort::new(
+                    chip.clone(),
+                    layouter.namespace(|| "2^64 - 1"),
+                    Some(scalar_fixed_short),
+                )?;
+                value_commit_v.mul(layouter.namespace(|| "[2^64 - 1]B"), &scalar_fixed_short)?
+            };
+            constrain_equal(
+                chip.clone(),
+                layouter.namespace(|| "[2^64 - 1]B"),
+                base_val,
+                scalar_fixed_short,
+                result,
+            )?;
+        }
+
+        // [-(2^64 - 1)]B
+        {
+            let scalar_fixed_short = -pallas::Scalar::from_u64(0xFFFF_FFFF_FFFF_FFFFu64);
+
+            let result = {
+                let scalar_fixed_short = ScalarFixedShort::new(
+                    chip.clone(),
+                    layouter.namespace(|| "-(2^64 - 1)"),
+                    Some(scalar_fixed_short),
+                )?;
+                value_commit_v.mul(layouter.namespace(|| "[-(2^64 - 1)]B"), &scalar_fixed_short)?
+            };
+            constrain_equal(
+                chip.clone(),
+                layouter.namespace(|| "[-2^64 - 1]B"),
+                base_val,
+                scalar_fixed_short,
+                result,
+            )?;
+        }
+
+        // There is a single canonical sequence of window values for which a doubling occurs on the last step:
+        // 1333333333333333333334 in octal.
+        // [0xB6DB_6DB6_DB6D_B6DC] B
```

Comment:
Test this value with both signs.

Reply:
Since these tests are refactored in #145, I have noted this comment there and will address it in that PR: https://github.com/zcash/orchard/pull/145/files#r670057072